pilot/pkg/networking/core/v1alpha3/httproute_test.go

```diff
@@ -83,6 +83,19 @@ func TestGenerateVirtualHostDomains(t *testing.T) {
 			},
 			want: []string{"foo.local.campus.net", "foo.local.campus.net:80"},
 		},
+		{
+			name: "k8s service with default domain",
+			service: &model.Service{
+				Hostname:     "echo.default.svc.cluster.local",
+				MeshExternal: false,
+			},
+			port: 8123,
+			node: &model.Proxy{
+				DNSDomain: "default.svc.cluster.local",
+			},
+			want: []string{"echo", "echo.default", "echo.default.svc", "echo.default.svc.cluster.local",
+				"echo:8123", "echo.default:8123", "echo.default.svc:8123", "echo.default.svc.cluster.local:8123"},
+		},
```

Comment:
sorry for all the back and forth

```go
		{
			name: "k8s service with custom domain 2",
			service: &model.Service{
				Hostname:     "google.local",
				MeshExternal: false,
			},
			port: 8123,
			node: &model.Proxy{
				DNSDomain: "foo.svc.custom.k8s.local",
			},
			want: []string{"google.local", "google.local:8123"},
		},
```

This test case would fail.

I think the problem here is we get the full dns domain (like `default.svc.cluster.local` from proxy) but we never consider the cluster domain (just `cluster.local`). We do have the info (see DomainSuffix) but its not passed down to this level I think. Unless its somehow in push context?

Reply:
No problem. I am happy to iterate on this. We would need to get this right in all cases :-)  I noticed that DomainSuffix part. We could probably get it from hostname? We could probably get it from hostname https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/conversion.go#L186 or add domainSuffix as one of the attributes in Service and access it here to be crystal clear.  I think putting in push context is difficult as options are not available when we build pushcontext unless we pass options to DiscoveryServer ?

Reply:
Do we need to find the domain of the Service or if the proxy? or both? They could be different, even in k8s, with multicluster for example.

If it's the proxy I think we can derive it most likely from the info we have (we know the namespace, so if we see <ns>.svc.<domain> pattern we can figure it out?)

Reply:
Thinking about this more - it seems proxy is sufficient. So should we just see if this is of the ".svc." pattern and derive the domain from there? I think that seems a reasonable fix.

Reply:
Changed. Check if it makes sense.